typescript/lib/services/service_context.ts

```diff
@@ -0,0 +1,129 @@
+import { ServiceID } from './service';
+import { StaticFileID } from './container_creation_config'; 
+import * as grpc from "grpc"; //TODO - this is needed for grpc callback, but this import is tentative 
+import * as constructors from "../constructor_calls";
+var path = require("path"); //TODO - I don't think this works
+
+// Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+class GeneratedFileFilepaths { //TODO (comment) - try interface if class doesn't work
+	absoluteFilepathOnTestsuiteContainer: string;
+	absoluteFilepathOnServiceContainer:   string;
+}
+
+class ServiceContext {
+    
+    client: bindingsJsGrpc.ApiContainerServiceClient;
+    serviceId: ServiceID;
+	ipAddress: string;
+	testVolumeMountpointOnTestsuiteContainer: string;
+	testVolumeMountpointOnServiceContainer: string;
+
+    constructor(
+        client: bindingsJsGrpc.ApiContainerServiceClient,
+		serviceId: ServiceID,
+		ipAddress: string,
+		testVolumeMountpointOnTestsuiteContainer: string,
+		testVolumeMountpointOnServiceContainer: string) {
+            this.client = client;
+            this.serviceId = serviceId;
+            this.ipAddress = ipAddress;
+            this.testVolumeMountpointOnTestsuiteContainer = testVolumeMountpointOnTestsuiteContainer;
+            this.testVolumeMountpointOnServiceContainer = testVolumeMountpointOnServiceContainer;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getServiceID(): ServiceID { 
+        return this.serviceId;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public getIPAddress(): string {
+        return this.ipAddress;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public execCommand(command: string[]): [number, [], Error] { //TODO - [int32, pointer to []byte (comment), error ] ; is is not []`type` like []string
+        let serviceId = this.serviceId;
+        let args = constructors.newGetExecCommandArgs(serviceId, command);
+        let request: grpc.requestCallback<bindingsJs.ExecCommandResponse>; //TODO - Tried to implement callback through examples on Google while also respecting the actual Callback function inside grpc
+
+        let resp, err = this.client.execCommand(args, request); //TODO - does error checking work here ; check the return type of execCommand, it's not an ExecCommandResponse ? 
+        
+        if (err != null) {
+            return [0, null, new Error( //TODO - Error type (do I need callback?)
+			    "An error occurred executing command '%v' on service '%v'", //TODO - I might not be able keep the '%v', but how I throw the actual error
+            )]; //TODO (comment) cannot return multipe values in typescript function, need to wrap in array or object
```

Comment:
I think that's fine - Javascript supports the same `returnVal1, returnVal2 = someFunction()` which is basically just as good